File: manual/src/test/java/examples/java/LoggingFileReviewer.java
package examples.java;

import org.approvej.approve.PathProvider;
import org.approvej.review.FileReviewResult;
import org.approvej.review.FileReviewer;
import org.approvej.review.FileReviewerProvider;
import org.approvej.review.ReviewResult;
import org.jspecify.annotations.NonNull;

public class LoggingFileReviewer implements FileReviewer, FileReviewerProvider {

  @Override
  public ReviewResult apply(PathProvider pathProvider) {
    System.out.println("Received: " + pathProvider.receivedPath());
    System.out.println("Approved: " + pathProvider.approvedPath());
    return new FileReviewResult(false);
  }

  @Override
  public @NonNull String alias() {
    return "logging";
  }

  @Override
  public FileReviewer create() {
    return new LoggingFileReviewer();
  }
}

File: manual/src/test/kotlin/examples/kotlin/LoggingFileReviewer.kt
package examples.kotlin

import org.approvej.approve.PathProvider
import org.approvej.review.FileReviewResult
import org.approvej.review.FileReviewer
import org.approvej.review.FileReviewerProvider
import org.approvej.review.ReviewResult

class LoggingFileReviewer : FileReviewer, FileReviewerProvider {
  override fun apply(pathProvider: PathProvider): ReviewResult {
    println("Received: " + pathProvider.receivedPath())
    println("Approved: " + pathProvider.approvedPath())
    return FileReviewResult(false)
  }

  override fun alias() = "logging"

  override fun create() = LoggingFileReviewer()
}
